app/components/BuildCard.tsx

```diff
@@ -109,13 +112,15 @@ export function BuildCard({ build, owner, canEdit = false }: BuildProps) {
       <div className="build__weapons">
         {weapons.map((weaponSplId) => (
           <div key={weaponSplId} className="build__weapon">
-            <Image
-              path={mainWeaponImageUrl(weaponSplId)}
-              alt={t(`weapons:MAIN_${weaponSplId}` as any)}
-              title={t(`weapons:MAIN_${weaponSplId}` as any)}
-              height={36}
-              width={36}
-            />
+            <Link to={`${BUILDS_PAGE}/${weaponIdSlug(weaponSplId)}`}>
```

Comment:
Define the value of `to` in urls.ts as a function. I'm trying to keep that file as source of truth for all the links in the project where feasible.

Reply:
Okay, I tried doing what you suggested, but by attempting to do so I think I broke the rules of React Hooks again & I'm having a hard time finding a way around this :(

See: https://github.com/Sendouc/sendou.ink/pull/1167/commits/9d5602994f96e74fdbf45193b58f65cac5e7a12c#diff-16131e5525021e86f0194c3c3b3867166462e434cb357fd3198e16d3b964fdd6

Basically, to make the new function in `urls.ts` that gets the weapon build page URL correctly, I need to disguise it as a React component by capitalizing the function name. No matter what, we'll have to end up using `useTranslation()` somewhere to properly render the weapon URL slug, but the problem is that there aren't any real React components in `urls.ts`.

Thing is that I know I shouldn't move this piece of code to something like a `.tsx` file because all we're expecting to get back from this function is text (and not a full-blown React component). The reliance on the `useTranslation()` hook is creating these odd problems & I'm just not seeing a solution that fully satisfies the rules of React hooks at this time, while also satisfying the constraint of centralizing the URL construction in `urls.ts`

Reply:
Oh, I just thought of something: putting the hook directly into `urls.ts`:

```js
export function useWeaponsBuildsPage(weaponId: MainWeaponId): string {
  const { t } = useTranslation("weapons");

  return mySlugify(t(`MAIN_${weaponId}`, { lng: "en" }));
}
```

Let me know if this is okay to do & I can do another refactor. Thanks again for your feedback!